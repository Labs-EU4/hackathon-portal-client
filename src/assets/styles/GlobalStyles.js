import { createGlobalStyle } from "styled-components";

import bg from '../images/Matrix-bg.jpg';

export const GlobalStyles = createGlobalStyle`
  /*MEYER RESET*/
  html, body, div, span, applet, object, iframe,
  h1, h2, h3, h4, h5, h6, p, blockquote, pre,
  a, abbr, acronym, address, big, cite, code,
  del, dfn, em, img, ins, kbd, q, s, samp,
  small, strike, strong, sub, sup, tt, var,
  b, u, i, center,
  dl, dt, dd, ol, ul, li,
  fieldset, form, label, legend,
  table, caption, tbody, tfoot, thead, tr, th, td,
  article, aside, canvas, details, embed,
  figure, figcaption, footer, header, hgroup,
  menu, nav, output, ruby, section, summary,
  time, mark, audio, video {
      margin: 0;
      padding: 0;
      border: 0;
      font-size: 100%;
      font: inherit;
      vertical-align: baseline;
  }

  /* make sure to set some focus styles for accessibility */
  :focus {
      outline: 0;
  }

  /* HTML5 display-role reset for older browsers */
  article, aside, details, figcaption, figure,
  footer, header, hgroup, menu, nav, section {
      display: block;
  }

  body {
      line-height: 1;
  }

  ol, ul {
      list-style: none;
  }

  blockquote, q {
      quotes: none;
  }

  blockquote:before, blockquote:after,
  q:before, q:after {
      content: '';
      content: none;
  }

  table {
      border-collapse: collapse;
      border-spacing: 0;
  }

  input[type=search]::-webkit-search-cancel-button,
  input[type=search]::-webkit-search-decoration,
  input[type=search]::-webkit-search-results-button,
  input[type=search]::-webkit-search-results-decoration {
      -webkit-appearance: none;
      -moz-appearance: none;
  }

  input[type=search] {
      -webkit-appearance: none;
      -moz-appearance: none;
      -webkit-box-sizing: content-box;
      -moz-box-sizing: content-box;
      box-sizing: content-box;
  }

  textarea {
      overflow: auto;
      vertical-align: top;
      resize: vertical;
  }

  audio,
  canvas,
  video {
      display: inline-block;
      *display: inline;
      *zoom: 1;
      max-width: 100%;
  }

  audio:not([controls]) {
      display: none;
      height: 0;
  }

  [hidden] {
      display: none;
  }

  html {
      font-size: 100%; /* 1 */
      -webkit-text-size-adjust: 100%; /* 2 */
      -ms-text-size-adjust: 100%; /* 2 */
  }

  a {
      -webkit-tap-highlight-color: transparent;
  }

  a:focus {
      outline: none;
  }

  a:active,
  a:hover {
      outline: 0;
  }

  img {
      border: 0; /* 1 */
      -ms-interpolation-mode: bicubic; /* 2 */
  }

  figure {
      margin: 0;
  }

  form {
      margin: 0;
  }

  fieldset {
      border: 1px solid #c0c0c0;
      margin: 0 2px;
      padding: 0.35em 0.625em 0.75em;
  }

  legend {
      border: 0; /* 1 */
      padding: 0;
      white-space: normal; /* 2 */
      *margin-left: -7px; /* 3 */
  }

  button,
  input,
  select,
  textarea {
      font-size: 100%; /* 1 */
      margin: 0; /* 2 */
      vertical-align: baseline; /* 3 */
      *vertical-align: middle; /* 3 */
  }

  button,
  input {
      line-height: normal;
      -webkit-tap-highlight-color: transparent;
  }

  button,
  select {
      text-transform: none;
  }

  button,
  html input[type="button"], /* 1 */
  input[type="reset"],
  input[type="submit"] {
      -webkit-appearance: button; /* 2 */
      cursor: pointer; /* 3 */
      *overflow: visible;  /* 4 */
  }

  button[disabled],
  html input[disabled] {
      cursor: default;
  }

  input[type="checkbox"],
  input[type="radio"] {
      box-sizing: border-box; /* 1 */
      padding: 0; /* 2 */
      *height: 13px; /* 3 */
      *width: 13px; /* 3 */
  }

  input[type="search"] {
      -webkit-appearance: textfield; /* 1 */
      -moz-box-sizing: content-box;
      -webkit-box-sizing: content-box; /* 2 */
      box-sizing: content-box;
  }

  input[type="search"]::-webkit-search-cancel-button,
  input[type="search"]::-webkit-search-decoration {
      -webkit-appearance: none;
  }

  button::-moz-focus-inner,
  input::-moz-focus-inner {
      border: 0px;
      padding: 0px;
      outline: 0px !important;
  }

  textarea {
      overflow: auto; /* 1 */
      vertical-align: top; /* 2 */
  }

  table {
      border-collapse: collapse;
      border-spacing: 0;
  }

  html,
  button,
  input,
  select,
  textarea {
      color: #222;
  }

  ::-moz-selection {
      background: #b3d4fc;
      text-shadow: none;
  }

  ::selection {
      background: #b3d4fc;
      text-shadow: none;
  }

  img {
      vertical-align: middle;
  }

  fieldset {
      border: 0;
      margin: 0;
      padding: 0;
  }

  textarea {
      resize: vertical;
  }

  .chromeframe {
      margin: 0.2em 0;
      background: #ccc;
      color: #000;
      padding: 0.2em 0;
  }

  /* Others ... */
  *, *::before, *::after {
      box-sizing: border-box;
  }

  :root {
      @import url('https://fonts.googleapis.com/css?family=Orbitron&display=swap');
      font-family: 'Orbitron', sans-serif !important;
      font-size: 62.5%;
      font-size: 57.5%;
      scroll-behavior: smooth;
      line-height: normal;
      --app-background: rgba(15,15,15,0.925);
      --app-text-color: #fff;
      --dash-side-margin: 20px;
      --app-padding: 21px;
      --border-radius: 5px;
      --border-color: rgba(255,255,255,.15);
      --border-color-low: rgba(255,255,255,.09);
      --header-height: 47px;
      --app-dropdown-bg: rgb(51,51,51);
      --app-message-bg: rgba(255,255,255,0.2);
      --icon-color: #fff;
      --a-show-full-duration: 0.3s;
      --a-show-full-curve: ease-out;
      --a-curve: ease;
      --a-fast: 0.15s;
      --a-default: all var(--a-fast) var(--a-curve);
  }

  body {
      max-width: 100vw;
      overflow: hidden;
      text-rendering: optimizeLegibility;
      font-weight: bold;
      -webkit-font-smoothing: antialiased;
      text-shadow: 0 1px 5px rgba(0,0,0,.1);
      line-height: 1.2;
      color: black;
      background-image: url(${bg}), linear-gradient(-225deg, #77FFD2 0%, #6297DB 48%, #1EECFF 100%);
      
      .dashboard-bg {
          height: 105vh;
          border: 3px solid red;
      }
  }

  .particles {
      position: fixed;
      top: 0;
      right: 0;
      bottom: 0;
      left: 0;
      z-index: 10;
  }
`;
