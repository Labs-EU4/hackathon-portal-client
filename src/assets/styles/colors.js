//COLORS
export const white = '#F9F9F9';
export const black = '#010038';

//RED COLOR
export const red = '#E34449';
export const brightRed = '#F34D21';

//YELLOW COLOR
export const dimYellow = '#ffdb00';

//GREY COLOR
export const grey = '#39444D';
export const textLightGrey = '#999';
export const textDarkGrey = '#666';
export const bgLightGrey = '#f0f0f0';
export const bgGrey = '#dfe3e6';
export const borderGrey = '#39444D';
// export const lightGrey = '#F5F5F5';

//GREEN COLOR
export const lightGreen = '#afeeee';
export const green = '#5aac44';
export const brightgreen = '#25d366';
export const darkGreen = '#1b9849';


// export const primary = '#04c8de'; //--> same as the above, good combo
// export const primary = '#F5F5F5'; //--> same as the above, good combo
// export const primary = '#FF3636'; //--> same as the above, good combo
// export const primary = '#47e4bb'; //--> same as the above, good combo

//PRIMARY COLOR
export const primaryLight = '#81DBFD';
export const primary = '#03AEEF';
export const primaryDark = '#03AEEF';
// export const primary = '#8CCAE9';
// export const primaryLight = '#04c8de';
// export const primary = '#40e0d0';

//SECONDARY COLOR
export const secondaryLight = '#F8C485';
export const secondary = '#f39422';
export const secondaryDark = '#DA7C0A';
// export const secondaryDark = '#F34D21';






export const backgroundBlack = '#222';
export const backgroundWhite = '#f4f4f4';
export const lightBlue = '#0ff';

// export const white = '#f4f7f6'; // --> GOOD WHITE
export const textWhite = '#F8F9FA';
export const primaryWhite = '#eff';
export const secondaryWhite = '#eee';
export const tertiaryWhite = '#ddd';
export const quaternaryWhite = '#ccc';
export const senaryWhite = '#aaa';




// export const black = '#000';
// export const black = '#1F1C05'; --> VERY GOOD BLACK SHADE
// export const black = '#020404 '; --> ALRIGHT BLACK!!!
// export const black = '#252525'; //--> SUPER GOOD BLACK
// export const black = '#202121';
export const softGreen = '#aec';
export const spaceBlue= '#03AEEF';
export const brightPurple= '#4444dd';
export const darkTwitterTheme = '#15202B';
export const lightDarkTwitterInputBg = '253341#';
export const fbCopyCatLightTheme = '#F3F4F6';
export const youTubeColorBlue = '#065FD4';
export const flickrBlue = '#1396E7';
export const btnBlue = '#0973FF';

//Add theme using styled components

//MAIN THEME COLORS
// export const primaryColor = '#BDE6EC';
// export const primaryColor = '#aec';
// export const primaryColor = '#ffdb00';
// export const primaryColor = '#6747C7'; //--> SUPER GOOD!!
// export const primaryColor = '#f8f4ff'; //--> GOOD WHITE VERSION!!
// export const primaryColor = '#fada5e'; //--> ROYAL YELLOW
// export const lightPrimaryColor = '#FEF33E'; //--> BRIGHT COOL YELLOW
// export const primaryColor = '#FFFF01'; //--> VERY BRIGHT YELLOW
// export const primaryColor = '#F4B75D'; //--> VERY BRIGHT YELLOW
// export const primaryColor = '#448ef6'; //--> VERY BRIGHT YELLOW
// export const primaryColor = '#6decb9'; //--> VERY BRIGHT YELLOW


// export const primaryColor = '#537ec5'; //--> super good color, mostly used theme!!!!
// export const turquoise = '#ff5edf'; //--> change to secondary color

export const tertiaryColor = '#5B5F97';
export const tertiaryColorDarker = '#484b77';

export const borderGrayTheme = '#39444D';

// export const primaryColor = '#1396E7'; //--> VERY GOOOD
// export const primaryColor = '#065FD4'; //--> VERY good blue, 2nd/3rd
// export const primaryColor = '#FFf'; //--> MOST USED YELLOW!!!! 
// export const primaryColor = '#B390EE';
// export const primaryColor = '#CC2D8A';
// export const primaryColor = '#46130B';
// export const primaryColor = '#FAD52D';

// export const turquoise = '#40e0d0 ';
// export const turquoise = '#4dd599'; --> VERY GOOD
// export const turquoise = '#47e4bb'; --> EVEN BETTER
// export const turquoise = '#3fc5f0';
//DARK-MODE TOGGLE COLORS
export const sunrise = '#ffc922';
export const yellowSun = '#ffdb00';
export const blueMoon = '#003366';
export const papayawhip = 'papayawhip';


// COOL COLORS 
export const newSecondaryColor = '#B390EE';
export const newTertiaryColor = '#B0EEEE';


// DARK MODE
export const MainBackgroundColor = '#29272F';
