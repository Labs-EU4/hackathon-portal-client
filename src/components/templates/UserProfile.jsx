import React from "react";
import styled from "styled-components";

import WideBody from "../atoms/WideBody";
import BodyContainer from "../atoms/BodyContainer";
import { H2, H3 } from "../atoms/Heading";
import { RowHead } from "../atoms/RowHead";
import { RowBody } from "../atoms/RowBody";
import { Column } from "../atoms/Column";
import { CardWide } from "../atoms/Card";
import Icon from "../atoms/Icon";
// import profileImg from "../../assets/profile-image.png";
import { media } from "../index";
import Button from "../atoms/Button";
import { Paragraph } from "../atoms/Paragraph";
// import mailIcon from "../../assets/Icon-mail-24px.png";
import EventCard from "../molecules/EventCard";

import { useRegisteredEvents } from "../../hooks";
import Spinner from "../molecules/Spinner";

export default function UserProfile({ initialState }) {
  const [data, loading] = useRegisteredEvents();
  const events = data?.body || [];
  return (
    <WideBody>
      <BodyContainerColumn>
        <RowHead>
          <H3>Your Profile</H3>
        </RowHead>

        <ProfileCardWide>
          <ProfileCard>
            <ProfileHead>
              {
                initialState.image_url &&
                 <ImageProfile src={JSON.parse(initialState.image_url[0])?.avatar}/> 
              }
              <Buttona color="green" anchor to="/dashboard/profile/edit">
                Edit profile
              </Buttona>
            </ProfileHead>
            <Column>
              <H2 style={{ marginBottom: 0 }}>{initialState.fullname}</H2>
              <Paragraph style={{ color: "gray" }}>
                @{initialState.username}
              </Paragraph>
            </Column>
            <Row>
              <Paragraph>{initialState.bio}</Paragraph>
            </Row>
            <Row>
              <Paragraph style={{ color: "blue", marginLeft: "3px" }}>
                {initialState.email}
              </Paragraph>
            </Row>
          </ProfileCard>

          <HackathonCard>
            <RowHead>
              <H3>Hackathon(s) you registered for</H3>
            </RowHead>
            {loading ? (
              <Spinner />
            ) : (
                <RowBody spacing="start">
                  {events.map(event => (
                    <EventCard key={event.id} event={event} />
                  ))}
                </RowBody>)}
          </HackathonCard>
        </ProfileCardWide>
      </BodyContainerColumn>
    </WideBody>
  );
}

const BodyContainerColumn = styled(BodyContainer)`
  flex-direction: column;
`;

export const ProfileCardWide = styled(CardWide)`
  max-width: 80%;
  min-width: 60%;
  justify-self: center;

  @media ${media.tablet} {
    max-width: 100%;
  }
`;

export const ProfileCard = styled.div`
  display: flex;
  flex-direction: column;
`;

export const HackathonCard = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: 8px;
`;

export const ImageProfile = styled.img`
  display: flex;
  width: 30%;
  border-radius: 50%;
`;

export const ProfileHead = styled.div`
  display: flex;
  justify-content: space-between;
  margin-bottom: 5px;
`;

export const Separator = styled.hr`
  border-top: 0;
  border-bottom: 1px solid #c3cfd9;
  margin: 0 0 20px 0;
`;

export const StyledIcon = styled.div`
  width: 20px;
  height: 23px;
`;

export const Row = styled.div`
  display: flex;
`;

const Buttona = styled(Button)`
  height: 15%;
  align-self: flex-end;
`;