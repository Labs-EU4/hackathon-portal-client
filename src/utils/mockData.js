export const initialState = {
  events: {
    data: [
      {
        id: 1,
        event_title: "EuroHack",
        event_description: "the best event hack hack hack macbook",
        creator_id: 1,
        start_date: "10/09/11",
        end_date: "08/07/13",
        location: "Europe",
        guidelines: "cool hekhkdsjfhsdkjf skdjfhskdjfhkjsdfh skjdfhksjdfh",
        participation_type: "Team",
        category_id: 1,
        requirements: [
          {
            video_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        ProfileImage: [
          {
            image_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        rubrics: [
          "Presentation",
          "Product Fit",
          "Functionality",
          "Innovation",
          "Product Design",
          "Extensibility"
        ]
      },
      {
        id: 2,
        event_title: "World",
        event_description: "number one hacker event in the world",
        creator_id: 1,
        start_date: "07/09/11",
        end_date: "08/07/13",
        location: "Asia",
        guidelines: "come well prepared,no cheating,no fighting",
        participation_type: "Team",
        category_id: 2
      }
    ],
    categories: [
      { id: 2, category_name: "Summer Hackton" },
      { id: 3, category_name: "World Hackton" },
      { id: 3, category_name: "Asia Hackton" },
      { id: 3, category_name: "Gamer's Hackton" },
      { id: 3, category_name: "Advanced Hackton" }
    ]
  },

  currentUser: {
    token: 347534574536,
    userId: 1,
    fullname: "Jake Tom",
    email: "8omemail@google.com",
    username: "Jake22",
    bio: "I like coding ",
    image: ""
  },
  submissions: [
    {
      id: 1,
      project_title: "Mock Project",
      participant_or_team_name: "Mock Team",
      event_id: 7,
      video_url: "",
      git_url: "https://www.google.com/",
      project_writeups: "No mock write up",
      submitted_by: 14,
      created_at: "2020-03-02",
      updated_at: "2020-03-02",
      average_rating: 0,
      acted_judges: 0,
      number_of_judges: 0
    }
  ],

  teams: [
    { id: 1, team_name: "The A team", team_lead: 2 },
    { id: 2, team_name: "Hacking central", team_lead: 3 },
    { id: 3, team_name: "blackhats", team_lead: 4 },
    { id: 4, team_name: "Spies", team_lead: 1 }
  ],
  participant_team_members: [
    {
      team_member: 3,
      team_id: 1
    },
    {
      team_member: 4,
      team_id: 1
    },
    {
      team_member: 5,
      team_id: 1
    }
  ]
};
