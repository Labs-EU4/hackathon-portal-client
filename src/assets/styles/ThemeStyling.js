import * as color from './variables/colors';

const size = {
    mobileS: '320px',
    mobileM: '375px',
    mobileL: '425px',
    tablet: '768px',
    laptop: '1024px',
    laptopL: '1440px',
    desktop: '2560px'
};

export const theme = {
    color: {
        primary: {
            light: '#81DBFD',
            regular: color.primary,
            // dark: color.primaryDark
        },
        secondary: {
            // light: color.secondaryLight,
            regular: color.secondary,
            // dark: color.secondaryDark
        },
        white: {
            regular: color.white,
            bg: color.bgWhite
        },
        black: {
            light: '#212121',
            regular: color.black,
        },
        red: {
            regular: color.red,
            bright: color.brightRed
        },
        blue: {
            light: color.lightBlue,
            regular: color.blue,
            bg: '#2196f3'
        },
        grey: {
            light: '#696969',
            regular: color.grey,
            dark: '#444',
            bg: color.bgGrey,
            border: 'rgba(14,19,24,.15)'
        },
        green: {
            light: color.lightGreen,
            regular: color.green,
            dark: color.darkGreen
        },
        success: color.success,
        warning: color.warning,
        danger: color.danger,
        info: color.info,
        link: {
            hover: 'rgba(14,19,24, .07)',
        }
    },
    fontSize: {
        smaller: "font-size: 1.2rem",
        small: "font-size: 1.4rem;",
        base: "font-size: 1.6rem;",
        medium: "font-size: 1.8rem",
        large: "font-size: 2rem",
        h1: "font-size: 7.5rem;",
        h2: "font-size: 5.5rem;",
        h3: "font-size: 3.5rem;",
        h4: "font-size: 2.5rem;",
        button: "font-size: 1.8rem;"
    },
    shadow: {
        text: `text-shadow: .5px .5px 1px ${color.black}`,
        text_links: `1px .75px 1.5px ${color.black}`,
        box: `box-shadow: 0 3px 20px 0 rgba(0, 0, 0, .2)`,
        card: `box-shadow: 1px .5px 3px rgba(0, 0, 0, .15)`,
        intenseBox: `box-shadow: 0 2.5px 10px ${color.black};`,
        filter: `filter: drop-shadow(.15px .15px 1.5px ${color.black});
        filter: drop-shadow(-.15px -.15px 1px ${color.black});`,
        btn: `2px 2px 3px ${color.black}`
    },
    flex: {
        center: 'display: flex; justify-content: center; align-items: center;',
        columnCenter: 'display: flex; flex-direction: column; justify-content: center; align-items: center;',
        column: 'display: flex; flex-direction: column;',

        custom: (justify='start', align='start', direction='row', wrap='no-wrap') => {
            return `
                display: flex; 
                flex-direction: ${direction}
                justify-content: ${justify}; 
                align-items: ${align};
                flex-wrap: ${wrap};
            `
        },
    },
    device: {
        mobileS: `(min-width: ${size.mobileS})`,
        mobileM: `(min-width: ${size.mobileM})`,
        mobileL: `(min-width: ${size.mobileL})`,
        tablet: `(min-width: ${size.tablet})`,
        laptop: `(min-width: ${size.laptop})`,
        laptopL: `(min-width: ${size.laptopL})`,
        desktop: `(min-width: ${size.desktop})`,
        desktopL: `(min-width: ${size.desktop})`
    }
};