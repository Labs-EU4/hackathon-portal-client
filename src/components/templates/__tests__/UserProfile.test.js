import React from "react";
import { Router } from "react-router";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render, cleanup } from "@testing-library/react";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom/extend-expect";
import UserProfile from "../UserProfile";
import { initialState } from "../../../utils/mockData";
import { theme } from "../../../assets/styles/ThemeStyling";
import { ThemeProvider } from "styled-components";
import { library } from '@fortawesome/fontawesome-svg-core';
import { fas } from '@fortawesome/free-solid-svg-icons';
import { far } from '@fortawesome/free-regular-svg-icons';
import { fab } from '@fortawesome/free-brands-svg-icons';

library.add(fas, far, fab);

const history = createMemoryHistory();

afterEach(cleanup);

let jestFeatures;
let mockStore;
let store;

beforeEach(() => {
  mockStore = configureStore();
  store = mockStore(initialState);
  jestFeatures = render(
    <Router history={history}>
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <UserProfile initialState={initialState} />
        </ThemeProvider>
      </Provider>
    </Router>
  );
});

describe("Component UserProfile.js text nodes renders properly", () => {
  it("asserts that the component renders properly", () => {
    expect(jestFeatures).toMatchSnapshot();
  });
  it("asserts that the text node on <H3> renders properly", () => {
    expect(jestFeatures.getByText("Your Profile")).toBeInTheDocument();
  });
  it("asserts that the text node on <Button> renders properly", () => {
    expect(jestFeatures.getByText("Edit profile")).toBeInTheDocument();
  });
  it("asserts that the text node on <H3> child of <HackathonCard> renders properly", () => {
    expect(
      jestFeatures.getByText("Hackathon(s) you registered for")
    ).toBeInTheDocument();
  });
});
