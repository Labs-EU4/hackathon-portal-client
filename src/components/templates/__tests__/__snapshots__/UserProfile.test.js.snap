// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component UserProfile.js text nodes renders properly asserts that the component renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-bdVaJa dErNmg"
        >
          <a
            class="sc-htoDjs jssEWI"
            href="/"
          >
            <img
              alt="Hackton - Organise hackathons"
              class="sc-bwzfXH fWCffE"
              src="Hackton-logo.png"
            />
          </a>
          <div
            class="sc-gzVnrw itepUc"
          >
            <div
              class="sc-EHOje bOqsil"
            >
              <button
                class="sc-htpNat maCfY sc-bZQynM ignRaC"
                color="grey"
                data-testid="label"
              >
                Log Out
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-bRBYWo efOACs"
        >
          <div
            class="sc-hzDkRC csxNDz sc-hEsumM gguRbo"
          >
            <div
              class="sc-fMiknA fZGKtL"
            >
              <h3
                class="sc-kAzzGY eOJFvg"
              >
                Your Profile
              </h3>
            </div>
            <div
              class="sc-kjoXOD sc-cHGsZl sc-ktHwxA eEUaxl"
            >
              <div
                class="sc-cIShpX gMSzSu"
              >
                <div
                  class="sc-iELTvK bSyfeV"
                >
                  <img
                    class="sc-feJyhm iCGigM"
                    src="profile-image.png"
                  />
                  <a
                    class="sc-ifAKCX dJBYdO sc-lhVmIH kVnYHQ"
                    color="green"
                    to="/dashboard/profile/edit"
                  >
                    Edit profile
                  </a>
                </div>
                <div
                  class="sc-caSCKo dZGQkd"
                >
                  <h2
                    class="sc-cSHVUG gUjuEA"
                    style="margin-bottom: 0px;"
                  />
                  <p
                    class="sc-ckVGcZ knXWAI"
                    style="color: gray;"
                  >
                    @
                  </p>
                </div>
                <div
                  class="sc-btzYZH jQEaJB"
                >
                  <p
                    class="sc-ckVGcZ knXWAI"
                  />
                </div>
                <div
                  class="sc-btzYZH jQEaJB"
                >
                  <div
                    class="sc-jwKygS dTTNdG"
                  />
                  <p
                    class="sc-ckVGcZ knXWAI"
                    style="color: blue; margin-left: 3px;"
                  />
                </div>
              </div>
              <div
                class="sc-kafWEX hSFtmy"
              >
                <div
                  class="sc-fMiknA fZGKtL"
                >
                  <h3
                    class="sc-kAzzGY eOJFvg"
                  >
                    Hackathon(s) you registered for
                  </h3>
                </div>
                <div
                  class="sc-fAjcbJ fvzHro"
                  spacing="center"
                >
                  <div
                    aria-busy="true"
                    class=""
                  >
                    <svg
                      aria-label="audio-loading"
                      height="100"
                      stroke="#2559a1"
                      viewBox="0 0 44 44"
                      width="100"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        fill="none"
                        fill-rule="evenodd"
                        stroke-width="2"
                      >
                        <circle
                          cx="22"
                          cy="22"
                          r="1"
                        >
                          <animate
                            attributeName="r"
                            begin="0s"
                            calcMode="spline"
                            dur="1.8s"
                            keySplines="0.165, 0.84, 0.44, 1"
                            keyTimes="0; 1"
                            repeatCount="indefinite"
                            values="1; 20"
                          />
                          <animate
                            attributeName="strokeOpacity"
                            begin="0s"
                            calcMode="spline"
                            dur="1.8s"
                            keySplines="0.3, 0.61, 0.355, 1"
                            keyTimes="0; 1"
                            repeatCount="indefinite"
                            values="1; 0"
                          />
                        </circle>
                        <circle
                          cx="22"
                          cy="22"
                          r="1"
                        >
                          <animate
                            attributeName="r"
                            begin="-0.9s"
                            calcMode="spline"
                            dur="1.8s"
                            keySplines="0.165, 0.84, 0.44, 1"
                            keyTimes="0; 1"
                            repeatCount="indefinite"
                            values="1; 20"
                          />
                          <animate
                            attributeName="strokeOpacity"
                            begin="-0.9s"
                            calcMode="spline"
                            dur="1.8s"
                            keySplines="0.3, 0.61, 0.355, 1"
                            keyTimes="0; 1"
                            repeatCount="indefinite"
                            values="1; 0"
                          />
                        </circle>
                      </g>
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-dnqmqq gkSNdn"
        >
          <p
            class="sc-VigVT kZDxgA"
          >
            International Crafters Â© 2020
          </p>
          <div>
            <a
              class="sc-iwsKbI sc-gqjmRU kqKCDB"
              href="#"
            >
              Privacy
            </a>
            <a
              class="sc-iwsKbI sc-gqjmRU kqKCDB"
              href="#"
            >
              Contacts
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-bdVaJa dErNmg"
      >
        <a
          class="sc-htoDjs jssEWI"
          href="/"
        >
          <img
            alt="Hackton - Organise hackathons"
            class="sc-bwzfXH fWCffE"
            src="Hackton-logo.png"
          />
        </a>
        <div
          class="sc-gzVnrw itepUc"
        >
          <div
            class="sc-EHOje bOqsil"
          >
            <button
              class="sc-htpNat maCfY sc-bZQynM ignRaC"
              color="grey"
              data-testid="label"
            >
              Log Out
            </button>
          </div>
        </div>
      </div>
      <div
        class="sc-bRBYWo efOACs"
      >
        <div
          class="sc-hzDkRC csxNDz sc-hEsumM gguRbo"
        >
          <div
            class="sc-fMiknA fZGKtL"
          >
            <h3
              class="sc-kAzzGY eOJFvg"
            >
              Your Profile
            </h3>
          </div>
          <div
            class="sc-kjoXOD sc-cHGsZl sc-ktHwxA eEUaxl"
          >
            <div
              class="sc-cIShpX gMSzSu"
            >
              <div
                class="sc-iELTvK bSyfeV"
              >
                <img
                  class="sc-feJyhm iCGigM"
                  src="profile-image.png"
                />
                <a
                  class="sc-ifAKCX dJBYdO sc-lhVmIH kVnYHQ"
                  color="green"
                  to="/dashboard/profile/edit"
                >
                  Edit profile
                </a>
              </div>
              <div
                class="sc-caSCKo dZGQkd"
              >
                <h2
                  class="sc-cSHVUG gUjuEA"
                  style="margin-bottom: 0px;"
                />
                <p
                  class="sc-ckVGcZ knXWAI"
                  style="color: gray;"
                >
                  @
                </p>
              </div>
              <div
                class="sc-btzYZH jQEaJB"
              >
                <p
                  class="sc-ckVGcZ knXWAI"
                />
              </div>
              <div
                class="sc-btzYZH jQEaJB"
              >
                <div
                  class="sc-jwKygS dTTNdG"
                />
                <p
                  class="sc-ckVGcZ knXWAI"
                  style="color: blue; margin-left: 3px;"
                />
              </div>
            </div>
            <div
              class="sc-kafWEX hSFtmy"
            >
              <div
                class="sc-fMiknA fZGKtL"
              >
                <h3
                  class="sc-kAzzGY eOJFvg"
                >
                  Hackathon(s) you registered for
                </h3>
              </div>
              <div
                class="sc-fAjcbJ fvzHro"
                spacing="center"
              >
                <div
                  aria-busy="true"
                  class=""
                >
                  <svg
                    aria-label="audio-loading"
                    height="100"
                    stroke="#2559a1"
                    viewBox="0 0 44 44"
                    width="100"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      stroke-width="2"
                    >
                      <circle
                        cx="22"
                        cy="22"
                        r="1"
                      >
                        <animate
                          attributeName="r"
                          begin="0s"
                          calcMode="spline"
                          dur="1.8s"
                          keySplines="0.165, 0.84, 0.44, 1"
                          keyTimes="0; 1"
                          repeatCount="indefinite"
                          values="1; 20"
                        />
                        <animate
                          attributeName="strokeOpacity"
                          begin="0s"
                          calcMode="spline"
                          dur="1.8s"
                          keySplines="0.3, 0.61, 0.355, 1"
                          keyTimes="0; 1"
                          repeatCount="indefinite"
                          values="1; 0"
                        />
                      </circle>
                      <circle
                        cx="22"
                        cy="22"
                        r="1"
                      >
                        <animate
                          attributeName="r"
                          begin="-0.9s"
                          calcMode="spline"
                          dur="1.8s"
                          keySplines="0.165, 0.84, 0.44, 1"
                          keyTimes="0; 1"
                          repeatCount="indefinite"
                          values="1; 20"
                        />
                        <animate
                          attributeName="strokeOpacity"
                          begin="-0.9s"
                          calcMode="spline"
                          dur="1.8s"
                          keySplines="0.3, 0.61, 0.355, 1"
                          keyTimes="0; 1"
                          repeatCount="indefinite"
                          values="1; 0"
                        />
                      </circle>
                    </g>
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-dnqmqq gkSNdn"
      >
        <p
          class="sc-VigVT kZDxgA"
        >
          International Crafters Â© 2020
        </p>
        <div>
          <a
            class="sc-iwsKbI sc-gqjmRU kqKCDB"
            href="#"
          >
            Privacy
          </a>
          <a
            class="sc-iwsKbI sc-gqjmRU kqKCDB"
            href="#"
          >
            Contacts
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
